catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/SearchController.java

```diff
@@ -181,26 +177,25 @@ public void executeQuery(final SearchRequest request, final ServerSession sessio
             Future future = entry.getValue();
             try {
                 long timeRemaining = Math.max(0, deadline - System.currentTimeMillis());
-
                 future.get(timeRemaining, TimeUnit.MILLISECONDS);
             } catch (InterruptedException e) {
-                LOGGER.info("Query interrupted for source " + sourceId, e);
-                failSource(request, session, search, sourceId);
+                LOGGER.info("Query interrupted for source {}", sourceId, e);
+                failSource(request, session, search, sourceId, e);
             } catch (TimeoutException e) {
-                LOGGER.info("Query timed out for source " + sourceId, e);
-                failSource(request, session, search, sourceId);
+                LOGGER.info("Query timed out for source {}", sourceId, e);
+                failSource(request, session, search, sourceId, e);
             } catch (ExecutionException e) {
-                LOGGER.warn("Query failed for source " + sourceId, e.getCause());
-                failSource(request, session, search, sourceId);
+                LOGGER.warn("Query failed for source {}", sourceId, e.getCause());
+                failSource(request, session, search, sourceId, e);
             }
         }
     }
 
     private void failSource(SearchRequest request, ServerSession session, Search search,
-            String sourceId) {
-        search.failSource(sourceId);
+            String sourceId, Exception cause) {
+        search.failSource(sourceId, cause);
         try {
-            pushResults(request.getId(), search.transform(request.getId()), session);
+            publishResults(request.getId(), search.transform(request.getId()), session);
```

Comment:
If you take a look in the `search.transform()` method, you will find the logic and method chain that actually constructs the JSON response.